src/single-spa-alpinejs.js

```diff
@@ -1,20 +1,46 @@
-// Opts are passed into single-spa-alpinejs when creating an application/parcel
+/**
+ * Opts are passed into single-spa-alpinejs when creating an application/parcel
+ * template: (required) main template as string
+ * xData: (optional) externally provided x-data as an object or function. The single spa env props are passed to this if its a function
+ * xInit: (optional) externally provided x-init as a function it takes an argument as the dom element ID of the application/parcel root element. This will only be used if xData is also provided
+ */
 const defaultOpts = {
-  // required
-  template: null,
+  template: null, // required
+  xData: undefined,
+  xInit: undefined,
 };
 
 export default function singleSpaAlpineJs(opts) {
   opts = Object.assign({}, defaultOpts, opts);
 
   const typeofTemplate = typeof opts.template;
+  const typeofXData = typeof opts.xData;
+  const typeofXInit = typeof opts.xInit;
 
   if (typeofTemplate !== "function" && typeofTemplate !== "string") {
     throw Error(
       `single-spa-alpinejs: opts.template must be provided as a string or a function that returns a promise that resolves with a string`
     );
   }
 
+  // xData can be optional , but when provided needs to be either a function or an object
+  if (
+    typeofXData !== "undefined" &&
```

Comment:
This line disallows `xData` from being passed in as `null`. It's often confusing to treat null and undefined differently. A truthiness check likely makes the most sense, I think. One thing to note is taht `typeof null` is `"object"`, which causes problems here. The code below solves this:

```suggestion
    opts.xData &&
```